File: Arrays/Sort Colors.rs
impl Solution {
    pub fn sort_colors(nums: &mut Vec<i32>) {
        let mut rc = 0;
        let mut bc = 0;

        for &num in nums.iter() {
            if num == 0 {
                rc += 1;
            } else if num == 1 {
                bc += 1;
            }
        }
        for i in 0..rc {
            nums[i] = 0;
        }
        for i in rc..rc+bc {
            nums[i] = 1;
        }
        for i in rc+bc..nums.len() {
            nums[i] = 2;
        }
    }
}

File: Binary Search/Sort Colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        zero_ptr = 0
        two_ptr = len(nums)-1
        mid = 0

        while mid <= two_ptr:
            if nums[mid] == 0:
                nums[mid], nums[zero_ptr] = nums[zero_ptr], nums[mid]
                zero_ptr += 1
                mid += 1
            elif nums[mid] == 2:
                nums[mid], nums[two_ptr] = nums[two_ptr], nums[mid]
                two_ptr -= 1
            else:
                mid += 1
